OpenRA.Mods.Common/Traits/Husk.cs

```diff
@@ -118,20 +124,23 @@ public void SetPosition(Actor self, WPos pos)
 			self.World.UpdateMaps(self, this);
 		}
 
-		public void AddedToWorld(Actor self)
+		void INotifyAddedToWorld.AddedToWorld(Actor self)
 		{
 			self.World.AddToMaps(self, this);
 		}
 
-		public void RemovedFromWorld(Actor self)
+		void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
 		{
 			self.World.RemoveFromMaps(self, this);
 		}
 
-		public void ModifyDeathActorInit(Actor self, TypeDictionary init)
+		void IDeathActorInitModifier.ModifyDeathActorInit(Actor self, TypeDictionary init)
 		{
 			init.Add(new FacingInit(Facing));
 		}
+
+		bool IEffectiveOwner.Disguised { get { return effectiveOwnerDisguised; } }
```

Comment:
Why don't you hardcode `true` here? Returning `self.Owner` from `IEffectiveOwner.Owner` shouldn't be a problem?

Reply:
Why should I hardcode it to `true`? `IEffectiveOwner.Owner` already returns `self.Owner` btw.

Reply:
Exactly because `IEffectiveOwner.Owner` returns `self.Owner`.  That method returns the correct result whether the source actor was disguised or not, so there is no need to store extra state to tell the game whether to query `self.Owner` from the actor, or `self.Owner` from `IEffective.Owner`.